graylog2-web-interface/src/stores/sidecars/CollectorsStore.js

```diff
@@ -177,8 +177,12 @@ const CollectorsStore = Reflux.createStore({
     CollectorsActions.copy.promise(promise);
   },
 
-  validate(name) {
-    const promise = fetch('GET', URLUtils.qualifyUrl(`${this.sourceUrl}/collectors/validate/?name=${name}`));
+  validate(name, collectorId) {
+    let url = `${this.sourceUrl}/collectors/validate/?name=${name}`;
+    if (collectorId) {
+      url += `&id=${collectorId}`;
```

Comment:
This works, but I'd rather use URI.js to handle the URL search parameters, now that the logic in there is a bit more complex. This is how it would look like:

```jsx
const search = {
  name: name,
};

if (collectorId) {
  search.id = collectorId;
}

const url = URI(`${this.sourceUrl}/collectors/validate`).search(search).toString();
const promise = fetch('GET', URLUtils.qualifyUrl(url));
```

What do you think?